### ISSUE TITLE
[Bug]: Defining TA_PORT creates invalid nginx configuration

### ISSUE THREAD

**Issue by anonamouslyginger** [tubearchivist/tubearchivist#436]
### I've read the documentation

- [X] I'm running the latest version of Tube Archivist and have read the [release notes](https://github.com/tubearchivist/tubearchivist/releases/latest).
- [X] I have read through the [wiki](https://github.com/tubearchivist/tubearchivist/wiki) and the [readme](https://github.com/tubearchivist/tubearchivist#installing-and-updating), particularly the [common errors](https://github.com/tubearchivist/tubearchivist#common-errors) section.

### Operating System

Linux

## Describe the bug

When overriding the `TA_PORT` a regular expression is executed to replace the port in the nginx configuration (see [here](https://github.com/tubearchivist/tubearchivist/blob/v0.3.3/tubearchivist/config/management/commands/ta_envcheck.py#L107-L122)).

This regular expression is matching and incorrectly replacing the `listen` directive in the `/etc/nginx/sites-available/default` file.

When running `docker exec -it tubearchivist bash -c 'cat /etc/nginx/sites-available/default'` without the `TA_PORT` override you get:
```
server {
    listen 8000;
```
But with the override you get:
```
server {
    8001;
```

Running `docker exec -it tubearchivist bash -c 'cat /var/log/nginx/*'` gives you the nginx error logs:
```
2023/02/18 23:46:55 [emerg] 17#17: unknown directive "8001" in /etc/nginx/sites-enabled/default:3
```

Checking for running ports shows that nginx is not running:
```
$ docker exec -it tubearchivist ss -tunlp

Netid                  State                   Recv-Q                  Send-Q                                   Local Address:Port                                    Peer Address:Port                 Process                                                               
udp                    UNCONN                  0                       0                                           127.0.0.11:53873                                        0.0.0.0:*                                                                                          
tcp                    LISTEN                  0                       100                                            0.0.0.0:8002                                         0.0.0.0:*                     users:(("uwsgi",pid=24,fd=3),("uwsgi",pid=20,fd=3))                  
tcp                    LISTEN                  0                       4096                                        127.0.0.11:36337                                        0.0.0.0:*  
```

## Steps To Reproduce

Define `TA_PORT` in your docker compose file.

## Expected behavior

The server should listen on the defined port.

### Relevant log output

```shell
#######################
#  Environment Setup  #
#######################

[1] checking expected env vars
    ✓ all expected env vars are set
[2] set default ES user
    ✓ ES user is set to elastic
[3] check TA_PORT overwrite
    ✓ TA_PORT changed to 8001
[4] check TA_UWSGI_PORT overwrite
    ✓ TA_UWSGI_PORT changed to 8002
[5] check ENABLE_CAST overwrite
    ENABLE_CAST is not set
[6] create superuser
    superuser already created
```

**Comment by birdwing** [comment]
Can confirm, got a bad gateway after updating to 0.3.3

Removing TA_PORT and allowing for the default 8000 port fixes the issue.

**Comment by anonamouslyginger** [comment]
Pull request created #437

**Comment by bbilly1** [comment]
Thanks a lot for looking into this and even creating the PR. :-)

I can see where this went wrong.

**Comment by anonamouslyginger** [comment]
No worries @bbilly1, thank you for the merge.

Do you have a timeline for when a patch version might be available?

**Comment by bbilly1** [comment]
I usually build an *unstable* image like once per week, then regular release like once per month. Depending what's ready to ship. Let's see how it goes. 

Maybe you'd be interested in tackling one of these things here #415? :-)

Also, it's quite easy to build with the existing dockerfile.

**Comment by bbilly1** [comment]
Thanks for fixing it, that's now published in v0.3.4.
